### PR TITLE
Initial f3d_loader and f3d_interactor API

### PR DESCRIPTION
**Pull request by mwestphal**
- Initial f3d_loader and f3d_interactor API

### CODE REVIEW

File: src/library/f3d_loader.h [hunk 1]
```diff
@@ -75,27 +73,40 @@ class F3DLoader
...
+   */
+  void InitializeRendering(std::string, bool offscreen, const void* iconBuffer, size_t inconLength);
 
-protected:
-  void InitializeRenderer();
+  /**
+   * Convenience method to get the interactor style
+   * TODO Remove
+   */
+  vtkF3DInteractorStyle* GetInteractorStyle();
```
Comment (mwestphal) [inline src/library/f3d_loader.h:104]: @Meakk : As of now, I do not see how we can remove this method while keeping full fonctionnality of F3D. Please advise

Comment (Meakk) [reply src/library/f3d_loader.h:104]: It's only used in `F3DStarter` to setup the observers, right?  
Then create a simple `f3d::interactor` holding this `vtkF3DInteractorStyle`. Move `f3d::interactor` to `F3DStarter` and call a `f3d::loader::setInterator`.

Comment (mwestphal) [reply src/library/f3d_loader.h:104]: F3DStarter is supported to go in the application, which is not supposed to link with VTK

**Comment by mwestphal** [comment]
@Meakk ready for initial review

**Review by Meakk — APPROVED** [review]
Looks good enough for now.

File: src/library/F3DStarter.cxx [hunk 2]
```diff
@@ -62,60 +61,54 @@ int F3DStarter::Start(int argc, char** argv)
...
-    newFilesCallback->SetCallback(
-      [](vtkObject*, unsigned long, void* clientData, void* callData)
+    // TODO Test this multiconfig behavior
+    this->Internals->Interactor.setKeyPressCallBack(
+      [this](int, std::string keySym) -> bool
       {
-        F3DStarter* starter = static_cast<F3DStarter*>(clientData);
-        vtkStringArray* filesArr = static_cast<vtkStringArray*>(callData);
-        for (int i = 0; i < filesArr->GetNumberOfTuples(); i++)
+        if (keySym == "Left")
```
[the remark this answers is not in the document]
Comment (mwestphal) [reply src/library/F3DStarter.cxx:68]: Of course, but they are defined in f3d::interactor

File: src/library/f3d_interactor.cxx [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+      // Make sure key symbols starts with an upper char (e.g. "space")
+      keySym[0] = std::toupper(keySym[0]);
+    }
+
+    if (self->KeyPressUserCallBack(keyCode, keySym))
+    {
+      return;
+    }
+
+    // No user defined behavior, use standard behavior
```
[the remark this answers is not in the document]
Comment (mwestphal) [reply src/library/f3d_interactor.cxx:50]: What would be the benefit ?

[the remark this answers is not in the document]
Comment (Meakk) [reply src/library/f3d_interactor.cxx:50]: It's cleaner. If the user sets the callback, the default behavior is overwritten.

File: src/library/F3DOptions.cxx [hunk 4]
```diff
@@ -732,67 +732,77 @@ void F3DOptionsParser::ConvertToNewAPI(const F3DOptions& oldOptions, f3d::option
...
+  newOptions->set("quiet", oldOptions.Quiet);
+  newOptions->set("verbose", oldOptions.Verbose);
+
+  // Loader/Loading
   newOptions->set("animation-index", oldOptions.AnimationIndex);
-  newOptions->set("axis", oldOptions.Axis);
+  newOptions->set("geometry-only", oldOptions.GeometryOnly);
+  newOptions->set("progress", oldOptions.Progress);
+  newOptions->set("camera-index", oldOptions.CameraIndex);
+  newOptions->set("color", oldOptions.SolidColor);
```
[the remark this answers is not in the document]
Comment (mwestphal) [reply src/library/F3DOptions.cxx:756]: so this is not using the f3d::options ? I'd rather  have a dedicated option category

**Comment by mwestphal** [comment]
CI is clean, coverage change is expected

**Pull request merged**